Code/Framework/AtomCore/CMakeLists.txt

```diff
@@ -6,11 +6,22 @@
 #
 #
 
+ly_add_target(
+    NAME AtomCore.API HEADERONLY
+    NAMESPACE AZ
+    FILES_CMAKE
+    AtomCore/atomcore_header_files.cmake
+    INCLUDE_DIRECTORIES
+        INTERFACE
+            .
+)
+
```

Comment:
AzFramework has the TerrainDataRequestBus which uses array_view. In this case, since we only need the header for the build dependency, I decided to split this into a header only and static target. I did this since AzFramework is a dependency to a lot of other libraries and executables and we needn't link AtomCore to every one of those(AtomCore needs to be a public dependency of AzFramework for reason mentioned below).

Reply:
This makes sense to me, but this will probably force the discussion over getting AZStd::span implemented, since I'm not sure how other folks will feel about this and the public AtomCore.API change below.

Reply:
Adding sig-build and sig-core to the review to get an opinion on this.